### PR TITLE
update eslint version and refactor all codebase

### PR DESCRIPTION
**Pull request by ricardofreitasrocha**
### JIRA link (if applicable) ###
N/A

### Change description ###
Update eslint and related configs

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
[x] Maybe
```

### CODE REVIEW

File: .eslintrc
Comment (gregs-cgi) [inline .eslintrc:1]: Prefer to delete the unused rules rather than comment them out.

**Review by annemariesteenkamp — APPROVED** [review]
I have had a brief test on this branch, I've tested some of the changed pages but not all of them. I can't see any problems so far and I feel it's OK to approve - If you want me to do further testing let me know.

File: .eslintrc [hunk 1]
```diff
@@ -1,86 +1,57 @@
...
+    "max-len": ["error", { "code": 120 }],
+    "no-var": "error",
+    "semi": ["error", "always"],
+    "comma-dangle": ["error", "always-multiline"],
+    "no-console": "warn",
+    "quotes": ["error", "single"],
+    "indent": ["error", 2, { "SwitchCase": 1 }],
     "eqeqeq": [2, "smart"],
-    "guard-for-in": 2,
+    "dot-location": [2, "property"],
```
Comment (gregs-cgi) [inline .eslintrc:21]: This rule is apparently deprecated (but clearly correct)

File: .eslintrc
Comment (gregs-cgi) [inline .eslintrc:1]: This file is inconsistent between using `rule: 'false'` or `rule: 'error'`, and `rule: 1`. Probably they should all use the string form for clarity.

File: .eslintrc [hunk 2]
```diff
@@ -1,86 +1,57 @@
...
-    "no-undefined": 2,
     "no-unused-vars": 1,
     "no-use-before-define": [2, {"functions": false}],
-
-    // Stylistic issues
-    "brace-style": 2,
-    "camelcase": 2,
+    "brace-style": "error",
+    "camelcase": "warn",
     "comma-spacing": [2, {"before": false, "after": true}],
```
Comment (gregs-cgi) [inline .eslintrc:27]: The args here are the defaults for this rule, so can be removed.

Comment (gregs-cgi) [reply .eslintrc:27]: Also, the rule is deprecated :D

**Comment by ricardofreitasrocha** [comment]
not doing now

**Pull request closed without merging**